m4/ax_check_pam.m4

```diff
@@ -26,34 +26,28 @@
 AC_DEFUN([AX_CHECK_PAM], [
     AC_ARG_WITH([pam],
         [AS_HELP_STRING([--with-pam=DIR],
-            [With libpam support, see ftp.redhat.com:/pub/pam])],
-        [
-            case "$withval" in
-            "" | y | ye | yes)
-                pamdirs="/usr/local /usr/lib /usr"  
-              ;;
-            n | no)
-                with_pam="no"
-              ;;
-            *)
-                pamdirs="$withval"
-              ;;
-            esac
-        ])
+            [With libpam support, see ftp.redhat.com:/pub/pam])])
     
-    if [ test "x${with_pam}" != "xno" ] ; then
-        PAM_LIBS="-lpam"
-        for pamdir in $pamdirs; do
-            AC_MSG_CHECKING([for pam_appl.h in $pamdir])
-            if test -f "$pamdir/include/security/pam_appl.h"; then
-                PAM_CFLAGS="-I$pamdir/include"
-                PAM_LDFLAGS="-L$pamdir/lib"
-                AC_MSG_RESULT([yes])
-                break
-            else
-                AC_MSG_RESULT([no])
-            fi
-        done
+    AS_IF([test "$with_pam" != "no"], [
+        AS_CASE(["$with_pam"],
+            [""|yes], [PKG_CHECK_MODULES([PAM], [pam], [pamdirs=],
+                        [pamdirs="/usr/local /usr/lib /usr"])],
```

Comment:
So you're swapping out the shell script's 'case' with AS_CASE, which in itself is fine and would improve compatibility; but you are eliminating a few of the case options, like what happens if you type --with-pam=ye, or --with-pam=n, or --with-pam=no. I'd like to see them added back with the AS_CASE.

Reply:
Passing `ye` or `n` to --with-pam makes no sense. Valid values should be yes, no, or a path. Trying to accept every possible misspelling seems silly to me.

Reply:
I don't think I care much about the 'ye' but the 'y' and 'n' or 'yes' and 'no' should at lest work (short-vs-long). Just hope that nobody like buildroot used a miss-typed 'ye' for yes.

Reply:
Also, the most common usage is to pass `--with-pam` which implies a value of `yes`, or `--without-pam` which implies a value of `no`.

Reply:
I added back handling for "ye", "y", and "n".